solar/solar/orchestration/graph.py

```diff
@@ -157,3 +160,23 @@ def report_topo(uid):
             data.get('end_time')])
 
     return report
+
+
+def wait_finish(uid, timeout, interval=3):
+    """Wait finish will periodically load graph and check if there is no
+    PENDING or INPROGRESS
+    """
+    start_time = time.time()
+
+    while start_time + timeout >= time.time():
+        dg = get_graph(uid)
+        summary = Counter()
+        summary.update({s.name: 0 for s in states})
+        summary.update([s['status'] for s in dg.node.values()])
+        yield summary
+        if summary[states.PENDING.name] + summary[states.INPROGRESS.name] == 0:
+            return
+        time.sleep(interval)
```

Comment:
Sleep shouldn't be in library, it should be on consumer side (CLI in this situation).